src/test/java/com/sharewalk/controller/UserControllerTest.java

```diff
@@ -0,0 +1,79 @@
+package com.sharewalk.controller;
+
+import com.sharewalk.dao.UserDAO;
+import com.sharewalk.model.User;
+import com.sharewalk.model.Walk;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class UserControllerTest {
+    @Rule
+    public MockitoRule mockito = MockitoJUnit.rule();
+
+    @Rule
+    public ExpectedException ex = ExpectedException.none();
+
+    @InjectMocks
+    private UserController instance;
+
+    @Mock
+    private UserDAO userDAO;
+
+    @Test
+    public void getUserWalksTest(){
+        Long userId = Long.valueOf(1L);
+        when(userDAO.listUserWalks(userId)).thenReturn(Arrays.asList(new Walk("walk 1", null, null), new Walk("walk 2", null, null) ));
```

Comment:
if controller only returns the list returned by service, you should check if the list if passed, like this:
List<Walk> walks = new ArrayList<>();
 when(userDAO.listUserWalks(userId)).thenReturn(walks)
ResponseEntity responseEntity = instance.getUserWalks(userId, null);
List result = ((List)responseEntity.getBody()).size());
assertEquals(walks,result)